themes/gohugoioTheme/layouts/_default/baseof.html

```diff
@@ -25,8 +21,20 @@
     {{ range .AlternativeOutputFormats -}}
     <link rel="{{ .Rel }}" type="{{ .MediaType.Type }}" href="{{ .Permalink | safeURL }}">
     {{ end -}}
-
-     <link href='{{ "dist/main.css" | relURL }}' rel='stylesheet' type="text/css" />
+    
+
+    {{ if (fileExists "./themes/gohugoioTheme/assets/output/css/app.css") -}}
```

Comment:
The motivation is that I have Webpack clean the assets/output directory and because Hugo is so fast there's sometimes a mismatch in build-times. I use this in a general way; i.e. once I solved the issue with this method I haven't tried to see if I need it on a specific instance. Happy to take it out if you don't want it.

Reply:
OK, I still don't understand it fully: What happens if Webpack wins this race? I assume you have to trigger a new change and cross fingers?

Reply:
I guess the above question was rhetorical, so I will add this:

I think it is the "webpack script" that should be responsible for the required locking here. I would be surprised if there isn't any "atomic save webpack plugin" out there that solves this -- a file rename is atomic on most OSes, so I think  a common pattern is to build to some other folder, and then do a rename when done.

Reply:
It would have to be a pretty slow Hugo build for Webpack to win. It's an imperfect system, but the best I've been able to come up with. It's conceivable to eliminate Webpack from this process entirely, but I think Webpack handles things like calling dependencies really well.